src/components/Navbar/Auth/PasswordRecoveryFinishForm.spec.js

```diff
@@ -0,0 +1,203 @@
+import React from "react";
+
+import { submitFinishPasswordRecover } from "../../../services/auth";
+
+import AuthModal from "./AuthModal";
+import { render, fireEvent, act } from "../../../test-utils";
+import Constants from "../../../utils/Constants";
+
+jest.mock("../../../services/auth");
+
+describe("Navbar - AuthModal", () => {
+    describe("render", () => {
+        it("Should not appear as default", () => {
+            const wrapper = render(<AuthModal initialPage={2} />);
+            const dialogTitle = wrapper.queryByRole("heading", { level: 2, name: "Recover Password" });
+            expect(dialogTitle).not.toBeInTheDocument();
+        });
+    });
+    describe("interaction", () => {
+        it("Should toggle the modal visibility when clicking Cancel button", () => {
+
+            const toggleAuthModal = jest.fn();
+
+            const wrapper = render(<AuthModal open toggleAuthModal={toggleAuthModal} initialPage={2} />);
+            const dialogTitle = wrapper.queryByRole("heading", { level: 2, name: "Recover Password" });
+            expect(dialogTitle).toBeInTheDocument();
+
+            const tokenField = wrapper.getByLabelText("Token");
+            expect(tokenField).toBeInTheDocument();
+
+            fireEvent.click(wrapper.getByText("Cancel"));
+
+            expect(toggleAuthModal).toHaveBeenCalledTimes(1);
+        });
+
+        it("Should change to the login page when clicking Login button", () => {
+
+            const toggleAuthModal = jest.fn();
+
+            const wrapper = render(<AuthModal open toggleAuthModal={toggleAuthModal} initialPage={2} />);
+            fireEvent.click(wrapper.getByText("Login"));
+
+            const dialogTitle = wrapper.queryByRole("heading", { level: 2, name: "Login" });
+            expect(dialogTitle).toBeInTheDocument();
+        });
+
+        it("Should request correctly and change to login page", async () => {
+
+            // Making sure that the login service allows the login
+            submitFinishPasswordRecover.mockImplementationOnce(() => true);
+
+            const toggleAuthModal = jest.fn();
+            const addSnackbar = jest.fn();
+
+            const { getByRole, getByLabelText, queryByRole } = render(
+                <AuthModal
+                    open
+                    toggleAuthModal={toggleAuthModal}
+                    updateSessionInfo={() => {}}
+                    initialPage={2}
+                    addSnackbar={addSnackbar}
+                />);
+
+            await act(async () => {
+                await fireEvent.change(getByLabelText("New Password"), { target: { value: "password123" } });
+            });
+            await act(async () => {
+                await fireEvent.change(getByLabelText("Token"), { target: { value: "some-token" } });
+            });
+            await act(async () => {
+                await fireEvent.click(getByRole("button", { name: "Recover Password" }));
+            });
```

Comment:
Couldn't these have been in the same `act(...)` call ?

Reply:
I pointed it out here but there are other places where this pattern is repeated.